packages/server/lib/plugins/index.js

```diff
@@ -61,7 +61,11 @@ const getChildOptions = (config) => {
 
   // To be removed on update to webpack >= 5.61, which no longer relies on
   // node's builtin crypto.hash function.
-  if (semver.satisfies(config.resolvedNodeVersion, '>=17.0.0')) {
+  // Node 17+ claims to support the new --openssl-legacy-provider option
+  // unconditionally, but it doesn't exist if Node was built against OpenSSL
+  // version 1.x.
```

Comment:
Can we rephrase this to be a little more informational? It took some digging to pin-point the original issue here.
Something like:
```js
// To be removed on update to webpack >= 5.61. Webpack 4 uses the `md4` cypro hash functional internally
// which has been marked as a legacy algorithm in OpenSSL 3.0.0. To use the legacy algoirthm, the Legacy Provider 
// must be used. Since Node 17+ uses OpenSSL 3 by default, Node added a new --openssl-legacy-provider
// option to enable OpenSSL 3 Legacy Provided. Since Node can be built against a different OpenSSL version,
// we need to verify OpenSSL 3+ is being used before passing the flag.
if (semver.satisfies(config.resolvedNodeVersion, '>=17.0.0')) {
```